index.html

```diff
@@ -176,18 +184,33 @@ <h4 class="modal-title">Extra Settings</h4>
               }
             })
 
+          var headers = ['artist', 'album', 'name', 'date'];
+          var includeMBID = $('#lastfm-include-mbid').prop('checked');
+          var includeHeaders = $('#lastfm-include-headers').prop('checked');
+
+          if(includeMBID) {
+            headers.push('mbid', 'artist_mbid', 'album_mbid');
+          }
+
           async.eachSeries(requests, function(item, callback){
             if(state.cancelled) return callback(false);
 
             state.status = "fetching page " + (item.i) + "/" + page_count;
             lastFM(item.data)
             .then(extractTracks)
             .then(function(tracks){
+              var r = tracks.map(function(d){
+                if(includeMBID) {
+                  return row(headers, d)
+                }
+                return row(headers, d)
+              })
+              .map(csv).join('\n') + '\n';
+
               var blb = new Blob([
-                tracks.map(function(d){
-                  return row(['artist', 'album', 'name', 'date'], d)
-                })
-                .map(csv).join('\n') + '\n']);
+                includeHeaders ? (headers.join(',') + '\n' + r) : r
```

Comment:
🤔 will this add the headers for every page of the output?

It might be easier to add the headers at the download handler instead:

https://github.com/benfoxall/lastfm-to-csv/blob/4d878050ee6cf15f9f2aaed1306dc1650f6eb36b/index.html#L144

⬆️ , could be something along the lines of ⬇️ 

```js
const header = includeHeaders ? [headers.join(',') + '\n'] : [];
var b = new Blob(header.concat(data), {type: 'text/csv'})
```

(though maybe there's a neater way to do it…)